datafusion/optimizer/src/filter_push_down.rs

```diff
@@ -80,6 +83,11 @@ impl State {
             .zip(predicates.1)
             .for_each(|(expr, cols)| self.filters.push((expr.clone(), cols.clone())))
     }
+
+    fn with_cnf_rewrite(mut self) -> Self {
```

Comment:
I think this should have some doccomments about what it is and when one it should/should not be set